src/ucp/api/ucp.h

```diff
@@ -494,6 +494,24 @@ typedef enum {
 } ucp_stream_recv_flags_t;
 
 
+/**
+ * @ingroup UCP_COMM
+ * @brief UCP operation fields and flags
+ *
+ * The enumeration allows specifying which fields in @ref ucp_op_info_t are
+ * present and operation flags are used. It is used to enable backward
+ * compatibility support.
+ */
+enum ucp_op_attr {
+    UCP_OP_ATTR_FIELD_REQUEST    = UCS_BIT(0), /**< request field */
+    UCP_OP_ATTR_FIELD_CB         = UCS_BIT(1), /**< cb field */
+    UCP_OP_ATTR_FIELD_USER_DATA  = UCS_BIT(2), /**< user_data field */
```

Comment:
can we always require `user_data`? Can be set to NULL by the user if it is not needed. Would help to avoid one of extra branches.